g3/generate/preview/cli.py

```diff
@@ -1,45 +1,39 @@
-import time
+from typing import Tuple
 
 import editor
 import inquirer
 from rich.console import Console
 from rich.markdown import Markdown
-from rich.progress import Progress, SpinnerColumn, TextColumn
 
-
-def display_selection(message, type, field="message") -> str:
-    options = [
-        "Submit",
-        "Edit",
-        "Regenerate",
-    ]
-
-    print(f"\n[+] Generated {type} {field}:\n")
-    console = Console()
-    md = Markdown(message, "github-dark")
-    console.print(md)
-
-    print()
-    questions = [
-        inquirer.List(
-            "selection",
-            message="Action",
-            choices=options,
-        )
-    ]
-
-    selection = inquirer.prompt(questions)["selection"]
-
-    if selection == "Edit":
-        message = editor.edit(contents=message).decode()
-    elif selection == "Regenerate":
-        with Progress(
-            SpinnerColumn(),
-            TextColumn("[progress.description]{task.description}"),
-            transient=True,
-        ) as progress:
-            progress.add_task(description="Regenerating...", total=None)
-            time.sleep(5)
-        # Placeholder for AI
-
-    return message
+options = ["Submit", "Edit", "Regenerate", "Cancel"]
+
+
+class Presenter:
+    @staticmethod
+    def present(message: str, type: str) -> Tuple[str, bool]:
+        if type not in ("pr", "commit"):
+            raise ValueError("Invalid message type")
+
+        print(f"\n[+] Generated {type} message:\n")
+        console = Console()
+        md = Markdown(message, "github-dark")
+        console.print(md)
+        print()
+        questions = [
+            inquirer.List(
+                "selection",
+                message="Action",
+                choices=options,
+            )
+        ]
+
+        selection = inquirer.prompt(questions)["selection"]
+        if selection == "Cancel":
+            exit(0)
+        elif selection == "Edit":
+            message = editor.edit(contents=message).decode()
+        elif selection == "Regenerate":
+            print("Regenerating..")
+            return message, True
```

Comment:
@tomtoump I tried the progress spinner in the message creator so that we keep it active till the message is (re)generated but I couldn't make it work. I will ping you to take a look together and potentially put it back in a follow-up commit